### ISSUE TITLE
Add ability to change voltage limits in plot

### ISSUE THREAD

**Issue by tomjholland** [ImperialCollegeLondon/PyProBE#104]

**Comment by brosaplanella** [comment]
I am happy to give a go at this one (if I understand correctly what you had in mind). Which API where you thinking of?

**Comment by tomjholland** [comment]
Hi Ferran, thanks and sorry for the vague issue description! I think this is a pretty simple issue:

So my `Plot` class uses plotly and this issue was to just fix the implementation of axes limits. The `Plot` class has `x_max`, `x_min`, `y_max`, `y_min` attributes which can be used to control the axes limits, but the actual plot adds some pad to this with the `x_range` and `y_range` properties, so the displayed limits are wider than the user requests:
![Screenshot 2024-09-27 at 10 07 35](https://github.com/user-attachments/assets/e209d936-ce17-4492-afae-77553d12db71)

The solution I think is probably just to add a setter for x_range and y_range to allow users to override the axes limits directly, and make the `*_min` and `*_max` attributes private.

**Comment by brosaplanella** [comment]
Cool! Will try to give it a go, if that's fine, but term is starting next week so if you don't hear from me and you need this done just go ahead.

**Comment by tomjholland** [comment]
Of course, no problem at all! I'm very grateful that you are keen to contribute and understand completely that you will have many other things to spend your time on 😃

**Comment by tomjholland** [comment]
#190 adds external plotting tools (matplotlib, hvplot and seaborn), which are expected to be the main user-facing plotting methods from now on. the `pyprobe.plot.Plot` class will remain to power the backend of the dashboard but no longer needs to be the focus of further development.
